packages/fuels-core/src/abi_encoder.rs

```diff
@@ -2,98 +2,152 @@ use crate::constants::{ENUM_DISCRIMINANT_WORD_WIDTH, WORD_SIZE};
 use crate::encoding_utils::{compute_encoding_width, compute_encoding_width_of_enum};
 use crate::errors::CodecError;
 use crate::{
-    pad_string, pad_u16, pad_u32, pad_u8, ByteArray, EnumSelector, EnumVariants, ParamType, Token,
+    pad_string, pad_u16, pad_u32, pad_u8, Bits256, ByteArray, EnumSelector, EnumVariants,
+    ParamType, Token,
 };
 use sha2::{Digest, Sha256};
-use std::slice;
 
 pub struct ABIEncoder {
-    pub function_selector: ByteArray,
-    pub encoded_args: Vec<u8>,
+    buffer: Vec<u8>,
 }
 
 impl ABIEncoder {
-    pub fn new() -> Self {
-        Self {
-            function_selector: [0; 8],
-            encoded_args: Vec::new(),
-        }
+    /// Encodes the function selector followin the ABI specs defined  ///
+    /// [here](https://github.com/FuelLabs/fuel-specs/blob/master/specs/protocol/abi.md)
+    pub fn encode_function_selector(fn_selector: &str) -> ByteArray {
+        let signature = fn_selector.as_bytes();
+        let mut hasher = Sha256::new();
+        hasher.update(signature);
+        let result = hasher.finalize();
+        let mut output = ByteArray::default();
+        (&mut output[4..]).copy_from_slice(&result[..4]);
+        output
     }
 
-    pub fn new_with_fn_selector(signature: &[u8]) -> Self {
-        Self {
-            function_selector: Self::encode_function_selector(signature),
-            encoded_args: Vec::new(),
+    /// Encodes `Token`s in `args` following the ABI specs defined
+    /// [here](https://github.com/FuelLabs/fuel-specs/blob/master/specs/protocol/abi.md)
```

Comment:
Do you mind using the permalink here? i.e. https://github.com/FuelLabs/fuel-specs/blob/1be31f70c757d8390f74b9e1b3beb096620553eb/specs/protocol/abi.md